### PR TITLE
Merged EmitterMixin additional parameters into options object.

### LINKED ISSUE #76: Merge EmiterMixins#on's ctx and prior args into options object
Having them separately is a problem for us because we want to add another option there in the DomEmitterMixin object – the `useCapture`.

```js
on( evtName, callback, options )
```

### PR DESCRIPTION
**Pull request by oskarwrobel**
Fixes #76 

Related:
- https://github.com/ckeditor/ckeditor5-ui/issues/79
- https://github.com/ckeditor/ckeditor5-engine/issues/593
- https://github.com/ckeditor/ckeditor5-typing/issues/44

### CODE REVIEW

**Comment by Reinmar** [comment]
Besides, I should review all 4 tickets at the same time, right?

**Comment by oskarwrobel** [comment]
> Besides, I should review all 4 tickets at the same time, right?

Yes.

File: src/emittermixin.js [hunk 1]
```diff
@@ -26,23 +26,27 @@ const EmitterMixin = {
...
 	 * the priority value the sooner the callback will be fired. Events having the same priority are called in the
 	 * order they were added.
-	 * @param {Object} [ctx] The object that represents `this` in the callback. Defaults to the object firing the
-	 * event.
+	 * @param {Object} [options.context] The object that represents `this` in the callback. Defaults to the object firing the event.
 	 * @method utils.EmitterMixin#on
 	 */
-	on( event, callback, priority = 'normal', ctx = null ) {
+	on( event, callback, options = {} ) {
+		if ( options.priority === undefined ) {
```
Comment (Reinmar) [inline src/emittermixin.js]: We must not modify objects that were passed to the methods.

Comment (Reinmar) [inline src/emittermixin.js]: Pity that we still can't use destructuring ;(

**Pull request merged**